### PR TITLE
Get events on iOS and android on par

### PR DESCRIPTION
**Pull request by pmogollons**
* Get events on iOS to be on par with android

### CODE REVIEW

**Comment by pmogollons** [comment]
@mieszko4 I got the events working on par with android. Will need to improve a bit android events later.

Im still strugling with onSinkMeetingVideoRequestUnmuteByHost and onSinkMeetingAudioRequestUnmuteByHost, this ones never get called. Can you take a look please?

File: ios/RNZoomUs.m [hunk 1]
```diff
@@ -600,14 +600,161 @@ - (void)onClickShareScreen:(UIViewController *)parentVC {
...
+- (void)onMeetingHostChange:(NSUInteger)userId {
+    [self sendEventWithName:@"MeetingEvent" params:@{
+    @"event": @"hostChanged",
+    @"userId": @(userId)
+  }];
+}
+
+- (void)onMeetingCoHostChange:(NSUInteger)userId {
+    [self sendEventWithName:@"MeetingEvent" params:@{
+    @"event": @"coHostChanged",
```
Comment (mieszko4) [inline ios/RNZoomUs.m:721]: 1. add `coHostChanged` event to `EVENTS.md` under `  //   IOS ONLY EVENTS`
2. move `myAudioStatusChanged` and `myVideoStatusChanged` out of `//   ANDROID ONLY EVENTS`

File: ios/RNZoomUs.m [hunk 2]
```diff
@@ -600,14 +600,161 @@ - (void)onClickShareScreen:(UIViewController *)parentVC {
...
+    MobileRTCMeetingUserInfo *userInfo = [ms userInfoByID:[ms myselfUserID]];
+
+    [self sendEventWithName:@"MeetingEvent" params:@{
+      @"event": @"myAudioStatusChanged",
+      @"userRole": [self getUserRole:[userInfo userRole]],
+      @"audioType": @([[userInfo audioStatus] audioType]),
+      @"isTalking": @([[userInfo audioStatus] isTalking]),
+      @"isMutedAudio": @((BOOL)([[userInfo audioStatus] audioType] == 2 ? YES : [[userInfo audioStatus] isMuted])),
+      @"isMutedVideo": @((BOOL)![[userInfo videoStatus] isSending]),
+    }];
```
Comment (mieszko4) [inline ios/RNZoomUs.m:680]: this repeats 3 times. on android we have
```
private void sendEvent(String name, String event, InMeetingUserInfo userInfo) {
```
So it would be great to have such abstraction on iOS, too

**Review by mieszko4 — COMMENTED** [review]
Looking great! I got some comments to consider.

I will check `onSinkMeetingVideoRequestUnmuteByHost` and `onSinkMeetingAudioRequestUnmuteByHost`

File: ios/RNZoomUs.m (hunk 2 above)
Comment (pmogollons) [reply ios/RNZoomUs.m:680]: Perfect, I was thinking about it but probably to tired to think about how to make it nicer.

**Comment by pmogollons** [comment]
Probably better to make this PR just to get event on android and iOS on par. Ill need to do the following.

- [ ] Add coHostChanged {userId} on android
- [ ] Updated hostChanged to add {userId} on android
- [ ] Check if we can improve the android onSinkMeetingActiveShare to not send screenShareStopped on every change

Something like this in iOS might be possible
```objective-c
if (userID == 0) {
  [self sendEventWithName:@"MeetingEvent" event:@"screenShareStopped"];
} else if ([ms isMyself:userID]){
  [self sendEventWithName:@"MeetingEvent" event:@"screenShareStarted"];
}
```

**Comment by mieszko4** [comment]
> Probably better to make this PR just to get event on android and iOS on par. Ill need to do the following.
> 
> * [ ]  Add coHostChanged {userId} on android
> * [ ]  Updated hostChanged to add {userId} on android
> * [ ]  Check if we can improve the android onSinkMeetingActiveShare to not send screenShareStopped on every change
> 
> Something like this in iOS might be possible
> 
> ```objc
> if (userID == 0) {
>   [self sendEventWithName:@"MeetingEvent" event:@"screenShareStopped"];
> } else if ([ms isMyself:userID]){
>   [self sendEventWithName:@"MeetingEvent" event:@"screenShareStarted"];
> }
> ```

Yes, that sounds good - this pr is already mergable and custom UI should be done in another one to easier track the changes/bugs etc.

**Comment by mieszko4** [comment]
@pmogollons 
> I will check `onSinkMeetingVideoRequestUnmuteByHost` and `onSinkMeetingAudioRequestUnmuteByHost`

I've tried to figure out this and my conclusion is that `onHostAskStartVideo !== onSinkMeetingVideoRequestUnmuteByHost`, i.e. they were implemented in different logic in mind....

`onHostAskStartVideo` is just a callback, no control to unmute in default UI

`onSinkMeetingVideoRequestUnmuteByHost` has `Accept` param based on which you decide whether you will accept the unmute. Perhaps it was mean to override the default confirmation dialog in Custom UI?
Or maybe it is a bug on iOS implementation - the docs are really hard to follow...

I believe it should trigger on Custom UI - based on https://devforum.zoom.us/t/ios-sdk-changename-dont-get-callback-event/21699/7

**Comment by mieszko4** [comment]
To confirm my theory even more I am asking myself the question: `Why would you want to listen to these events on ios if you cannot control the behaviour in the Default UI`?

On Android you can set `no_unmute_confirm_dialog = true` in which case you must implement `onHostAskStartVideo` to have an effect "to handle the notifications or actions, otherwise pressing the unmute/ask to start video by the host will have no impact." (See https://devforum.zoom.us/t/disable-notification-onhostaskstartvideo/7772)

For iOS there is no such option to disable the confirm dialog so this is probably why it does not trigger (there is no need for that to trigger)

I suggest to just write a simple comment in `EVENTS.md` for these events for now. We should merge this pr, then when Custom UI will be done we can verify if these events work :)

**Comment by pmogollons** [comment]
Interesting, so Ill just leave stubs for those methods in iOS for now.

**Comment by pmogollons** [comment]
I want to propose these last changes to the EVENTS docs, it makes it easier to understand, updates the listener names and uses NativeModules to fix an issue with RN 0.65.

**Comment by pmogollons** [comment]
Fixed #143

**Comment by pmogollons** [comment]
Im getting an issue on android. 

```java
    java.lang.NullPointerException: Attempt to invoke interface method 'us.zoom.sdk.InMeetingShareController us.zoom.sdk.InMeetingService.getInMeetingShareController()' on a null object reference
```

**Comment by pmogollons** [comment]
I was checking PR #40 and it looks like most of it is already implemented by this and other PR's, we should close that PR.

**Comment by mieszko4** [comment]
> Im getting an issue on android.
> 
> ```java
>     java.lang.NullPointerException: Attempt to invoke interface method 'us.zoom.sdk.InMeetingShareController us.zoom.sdk.InMeetingService.getInMeetingShareController()' on a null object reference
> ```
> 
> The issue was mostly on my apps side. I was calling stopVideo before the meeting was started. But this raises the question, should we check if meeting has started?

`I was calling stopVideo before the meeting was started` - I do not understand...
Create an issue please so we do not get lost here :)

**Comment by mieszko4** [comment]
> I was checking PR #40 and it looks like most of it is already implemented by this and other PR's, we should close that PR.

I've closed that - I took a review on that again - it has some useful code/patterns for reference - I've created https://github.com/mieszko4/react-native-zoom-us/issues/147

**Review by mieszko4 — APPROVED** [review]
thnx @pmogollons

File: docs/EVENTS.md [hunk 3]
```diff
@@ -6,102 +6,114 @@ and meeting events as follows:
 Remember that you should add the listener after the sdk has been initialized successfully.
 
 ```javascript
-import { ZoomEmitter } from 'react-native-zoom-us';
+import { NativeModules } from 'react-native';
 
-const zoomEmitter = new NativeEventEmitter(ZoomEmitter);
+const zoomEmitter = new NativeEventEmitter(NativeModules.ZoomUs);
```
[the remark this answers is not in the document]
Comment (pmogollons) [reply docs/EVENTS.md:11]: Im not sure the reason, I was having this issue and didnt think that was the reason because as you say, is the same. But after doing that change I stopped getting a weird warning for NativeEventEmitter in the console.

[the remark this answers is not in the document]
Comment (mieszko4) [reply docs/EVENTS.md:11]: I took a look on that and I've fixed the warnings on android in https://github.com/mieszko4/react-native-zoom-us/pull/150
